pandas/formats/format.py

```diff
@@ -1951,7 +1951,11 @@ def _format_hierarchical_rows(self):
                 for spans, levels, labels in zip(level_lengths,
                                                  self.df.index.levels,
                                                  self.df.index.labels):
-                    values = levels.take(labels)
+
+                    values = levels.take(labels,
+                                         allow_fill=levels._can_hold_na,
```

Comment:
If I leave out the `allow_fill=levels._can_hold_na`, I get:

```
In [1]: df = DataFrame({'x': np.random.uniform(0, 1, 3), 'y': np.random.uniform(0, 1, 3)})

In [2]: df['A'] = [1,1,3]

In [3]: df['B'] = ['a', np.nan, 'b']

In [4]: df = df.set_index(['A', 'B'])

In [5]: df
Out[5]:
              x         y
A B
1 a    0.335609  0.521433
  NaN  0.531680  0.265201
3 b    0.910320  0.520158

In [6]: df.to_excel('test.xlsx')
---------------------------------------------------------------------------
ValueError: Unable to fill values because Int64Index cannot contain NA
```

But the problem is that the test frame only contains floats. @mpuels can you update the test to also include an integer index case?